src/huggingface_hub/hf_api.py

```diff
@@ -128,12 +128,13 @@ def repo_type_and_id_from_hf_id(hf_id: str):
     return repo_type, namespace, repo_id
 
 
-class RepoObj:
+class RepoFile:
```

Comment:
This is a breaking change, the statement `import RepoObj from huggingface_hub.hf_api` has no indication that `RepoObj` is a private class and it's being renamed here.

If users are not supposed to import from `huggingface_hub.hf_api`, then we should rename this file to `_hf_api.py` and deprecate the old import pattern and let users import only from `huggingface_hub` root module.

If you think this class is never used by any of our users then sure, we can rename it, but we should then rename the file to make sure people import things that we think are *public* API.

Reply:
yes I'm 99.9% sure no user ever used this, it hasn't been used for a long time.

That being said, no objection to renaming the file to `_hf_api.py` if it's the canonical way to make a internal file "private" (not consistent w/ what we do in other libraries but no big deal)

Reply:
Cool, we can leave the renaming of the file to a separate PR.